tests/pxcentral/pxcentral_test.go

```diff
@@ -0,0 +1,535 @@
+				logrus.Infof("Got px-backup-ui endpoint: %s", endpoint)
+			})
+
+			Step("Getting OIDC client secret", func() {
+				secretObj, err := core.Instance().GetSecret("pxc-backup-secret", context.ScheduleOptions.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				secretData, exist := secretObj.Data["OIDC_CLIENT_SECRET"]
+				Expect(exist).To(Equal(true))
+				oidcSecret = base64.StdEncoding.EncodeToString(secretData)
+				logrus.Infof("Got OIDC client secret: %s", oidcSecret)
+			})
+
+			Step("Adding extra values to helm ConfigMap", func() {
+				configMap, err := core.Instance().GetConfigMap(appName, "default")
+				Expect(err).NotTo(HaveOccurred())
+
+				configMap.Data[k8s.HelmExtraValues] = fmt.Sprintf("pxmonitor.pxCentralEndpoint=%s,pxmonitor.oidcClientSecret=%s",
+																  endpoint,
+																  oidcSecret)
+				configMap, err = core.Instance().UpdateConfigMap(configMap)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("Got extra helm values for px-monitor: %s", configMap.Data[k8s.HelmExtraValues])
+			})
+
+			Step("Install px-monitor", func() {
+				err := Inst().S.AddTasks(context, *monitorOptions)
+				Expect(err).NotTo(HaveOccurred())
+
+				ValidateContext(context)
+				logrus.Infof("Successfully validated specs for px-monitor")
+
+				// remove extra values in config map
+				if !configMutable {
+					configMap, err := core.Instance().GetConfigMap(appName, "default")
+					configMap.Data[k8s.HelmExtraValues] = ""
+					configMap, err = core.Instance().UpdateConfigMap(configMap)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+		})
+	}
+
+	return context
+}
+
+// deleteAndWait waiting for resources to be deleted
+func deleteAndWait(resource interface{}) error {
+	if obj, ok := resource.(*corev1.Namespace); ok {
+		t := func() (interface{}, bool, error) {
+			err := core.Instance().DeleteNamespace(obj.Name)
+
+			if err != nil && strings.Contains(err.Error(), "not found") {
+				return nil, false, err
+			}
+			return nil, true, errors.New(fmt.Sprintf("namespace %s not deleted", obj.Name))
+		}
+
+		_, err := task.DoRetryWithTimeout(t, defaultTimeout, defaultRetryInterval)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("not found"))
+		logrus.Infof("namespace %s deleted", obj.Name)
+	} else if obj, ok := resource.(*batchv1.Job); ok {
+		t := func() (interface{}, bool, error) {
+			err := batch.Instance().DeleteJob(obj.Name, obj.Namespace)
+
+			if err != nil && strings.Contains(err.Error(), "not found") {
+				return nil, false, err
+			}
+			return nil, true, errors.New(fmt.Sprintf("job %s not deleted", obj.Name))
+		}
+
+		_, err := task.DoRetryWithTimeout(t, defaultTimeout, defaultRetryInterval)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("not found"))
+		logrus.Infof("job %s deleted", obj.Name)
+	} else {
+		return errors.New("resource type not implemented")
+	}
+
+	return nil
+}
+
```

Comment:
we can move this implementation under the scheduler driver, just add two methods DeleteNamespace and DeleteJob and then call them in your test